subprojects/docs/docs.gradle

```diff
@@ -266,8 +267,12 @@ task dslHtml(type: Docbook2Xhtml) {
     ext.entryPoint = "$destDir/index.html"
 }
 
+task checkSectionIds(type: UserGuideSectionVerifierTask) {
+    docbookFiles = fileTree(userguideSrcDir) { include "**/*.xml" }
+}
+
 // This is used in the distribution and for the online version
-task userguideDocbook(type: UserGuideTransformTask, dependsOn: [samples, samplesDocbook, defaultImports]) {
+task userguideDocbook(type: UserGuideTransformTask, dependsOn: [checkSectionIds, samples, samplesDocbook, defaultImports]) {
```

Comment:
We'll likely want to check IDs as the first thing in the list of task dependencies so we can fail early. We could use a `mustRunAfter` statement to define the task execution order.